### PR TITLE
[Issue-194][Feature] Support spark 3.2.0

### PR DESCRIPTION
**Pull request by leixm**
### What changes were proposed in this pull request?
support spark 3.2.0

### Why are the changes needed?
After SPARK-36892, MapOutputTrackerWorker#getMapSizesByExecutorId has been modified to include method names and return parameter types, Java.lang.NoSuchMethodException will be thrown when the RssShuffleManager#getExpectedTasksByExecutorId method is called, The version of our production environment is 3.2.0.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
reused intergration-test

### CODE REVIEW

**Comment by leixm** [comment]
We need to add a "-Pspark3.2.0" pipeline.

**Comment by jerqi** [comment]
You can add the pipeline in https://github.com/apache/incubator-uniffle/blob/master/.github/workflows/parallel.yml

File: client-spark/spark3/src/main/java/org/apache/spark/shuffle/RssShuffleManager.java [hunk 1]
```diff
@@ -483,8 +484,21 @@ private Roaring64NavigableMap getExpectedTasksByExecutorId(
                     shuffleId,
                     startPartition,
                     endPartition);
-      } catch (Exception ee) {
-        throw new RuntimeException(ee);
+      } catch (Exception ignored1) {
+        try {
+          // attempt to use Spark 3.2.0's API
```
Comment (jerqi) [inline client-spark/spark3/src/main/java/org/apache/spark/shuffle/RssShuffleManager.java:489]: Could we add more comments like
```
Each Spark release will be versioned: [MAJOR].[FEATURE].[MAINTENANCE].  Usually we only need to adapt [MAJOR].[FEATURE]. Unfortunately, some interfaces were removed wrongly in Spark 3.2.0. And they were added by Spark 3.2.1. So we need to adapt Spark 3.2.0 here
```
```

**Comment by jerqi** [comment]
Modify the related the document at the same time.

**Comment by leixm** [comment]
All done, Can you help review please? @jerqi

**Review by jerqi — APPROVED** [review]
LGTM, thanks @leixm

**Comment by leixm** [comment]
Thank you for your patience and kindness. @jerqi

**Pull request merged**